pkg/agent/client/client.go

```diff
@@ -109,17 +111,26 @@ func (c *client) FetchUpdates(ctx context.Context, req *node.FetchX509SVIDReques
 			break
 		}
 		if err != nil {
-			logrus.Errorf("failed to consume entire SVID update stream: %v", err)
+			c.c.Log.Errorf("failed to consume entire SVID update stream: %v", err)
 			c.release(nodeConn)
 			return nil, err
 		}
 
 		if resp.SvidUpdate == nil {
-			logrus.Warn("empty update in SVID update stream")
+			c.c.Log.Warn("empty update in SVID update stream")
 			continue
 		}
 
 		for _, re := range resp.SvidUpdate.RegistrationEntries {
+			if err := validateRegistrationEntry(re); err != nil {
+				c.c.Log.WithFields(logrus.Fields{
+					telemetry.RegistrationID: re.EntryId,
+					telemetry.SPIFFEID:       re.SpiffeId,
+					telemetry.Selectors:      strconv.Itoa(len(re.Selectors)),
```

Comment:
The selectors field IIRC has historically had the actual selectors as its value. The name `selectors` also feels kind of weird when paired with a number, at least in the context of how its used elsewhere in the project. One option is to use a new field name e.g. `selector_count`... but, I think I'd prefer to just see this left out entirely... thoughts?

Reply:
That's a good point. From a debuggability perspective, I'd rather keep it (and change it to logging the selectors themselves) since in the particular case where this was happening, both the registration id and spiffe id would have been empty. Only the selectors gave a clue as to what was happening. This shouldn't ever happen in the normal case; I'd rather give them the information they need to rectify it when it does.

Reply:
👍 fair enough, logging the list sounds fine to me